library/spdm_requester_lib/libspdm_req_get_certificate.c

```diff
@@ -158,11 +160,28 @@ return_status try_spdm_get_certificate(IN void *context, IN uint8_t slot_id,
             status = RETURN_DEVICE_ERROR;
             goto done;
         }
+        if (spdm_request.offset == 0) {
+            total_responder_cert_chain_buffer_length = spdm_response.portion_length + spdm_response.remainder_length;
```

Comment:
```
        if (spdm_response.portion_length >
            LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN) {
            status = RETURN_DEVICE_ERROR;
            goto done;
        }
```
can be moved inside here and use `total_responder_cert_chain_buffer_length`.

Reply:
As we can see, the logic` else if (spdm_request.offset + spdm_response.portion_length + spdm_response.remainder_length != total_responder_cert_chain_buffer_length)` is strong.

So I have deleted the check:
```
        if (spdm_response.portion_length >
            LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN) {
            status = RETURN_DEVICE_ERROR;
            goto done;
        }
```
And is my way right? Look forward to receiving your reply. Thanks.

Reply:
I dont think it is right.
"spdm_response.portion_length <= LIBSPDM_MAX_CERT_CHAIN_BLOCK_LEN" cannot be enforced by total length check.

In addition, we need add another check
"spdm_response.portion_length > 0" to ensure we always move forward.

Reply:
Yes. I have added the check.